### ISSUE TITLE
Use opencv as a solver in meta-training?

### ISSUE THREAD

**Issue by biantaotiyan** [icoz69/DeepEMD#7]
Hi, 

Thanks for your inspring work! However, I have a question.

I notice that in your readme file of the project, a command is provided to meta-train the model, after pre-training is done, where opencv is specified as the solver:
===============
#use opencv solver (about 8GB memory)
$ python train_meta.py -deepemd fcn -shot 1 -way 5 -solver opencv -gpu 0,1,2,3
===============
Nevertheless, in your paper, 5.1 Implementation Details, Training paragraph, it's claimed that QPTH is used during meta-training: “At training time, we use the GPU accelerated convex optimization solver QPTH [76] to solve the Linear Programming problem in our network and compute gradients for back-propagation.”

So, my question is:
1) Is opencv available during meta-training?
2) If both opencv and QPTH are available during meta-training? Does that mean for opencv, additional coding work has been done in the project to calculate gradients for back-propagation, like what has been done in QPTH?

Best Regards,
Neil

**Comment by biantaotiyan** [comment]
> Hi,
> 
> Thanks for your inspring work! However, I have a question.
> 
> # I notice that in your readme file of the project, a command is provided to meta-train the model, after pre-training is done, where opencv is specified as the solver:
> # #use opencv solver (about 8GB memory)
> $ python train_meta.py -deepemd fcn -shot 1 -way 5 -solver opencv -gpu 0,1,2,3
> Nevertheless, in your paper, 5.1 Implementation Details, Training paragraph, it's claimed that QPTH is used during meta-training: “At training time, we use the GPU accelerated convex optimization solver QPTH [76] to solve the Linear Programming problem in our network and compute gradients for back-propagation.”
> 
> So, my question is:
> 
> 1. Is opencv available during meta-training?
> 2. If both opencv and QPTH are available during meta-training? Does that mean for opencv, additional coding work has been done in the project to calculate gradients for back-propagation, like what has been done in QPTH?
> 
> Best Regards,
> Neil
